middleware.go

```diff
@@ -76,6 +76,11 @@ func parseRequestHeader(c *Client, r *Request) error {
 		hdr[k] = append(hdr[k], r.Header[k]...)
 	}
 
+	// Allow the Request to override the Authorization header set by the Client
```

Comment:
I have analyzed the v0.13 and latest version. Fix for across headers.

Remove this if condition and update your PR as follows.

Update [Line #75 to #77](https://github.com/go-resty/resty/blob/master/middleware.go#L75-L77) to following-  
```go
for k := range r.Header {
	hdr.Del(k)
	hdr[k] = append(hdr[k], r.Header[k]...)
}
```